File: aoc-2015/src/test/java/com/capital7software/aoc2015/days/Day23Test.java
package com.capital7software.aoc2015.days;

import com.capital7software.aoc.aoc2015.days.Day23;
import com.capital7software.aoc.lib.AdventOfCodeTestBase;
import org.junit.jupiter.api.BeforeEach;
import org.junit.jupiter.api.Test;

import java.util.logging.Logger;

import static org.junit.jupiter.api.Assertions.assertEquals;

class Day23Test extends AdventOfCodeTestBase {
    private static final Logger LOGGER = Logger.getLogger(Day23Test.class.getName());

    @BeforeEach
    void setUp() {
        var instance = new Day23();
        setupFromFile(instance.getDefaultInputFilename());
    }

    @Test
    void runProgramAndGetValueInRegister() {
        var instance = new Day23();
        int expected = 2;
        var actual = instance.runProgramAndGetValueInRegister(lines, "a");
        assertEquals(expected, actual, "The value in register a " +
                "is not what was expected: " + expected);
    }

    @Test
    void runProgramWithStartingRegisterAndValueAndGetValueInRegister() {
        var instance = new Day23();
        int expected = 7;
        var actual = instance.runProgramWithStartingRegisterAndValueAndGetValueInRegister(lines, "a", 1L, "a");
        assertEquals(expected, actual, "The value in register a " +
                "is not what was expected: " + expected);
    }

    @Override
    protected Logger getLogger() {
        return LOGGER;
    }
}

File: aoc-2016/src/test/kotlin/com/capital7software/aoc/aoc2016aoc/days/Day23Test.kt
package com.capital7software.aoc.aoc2016aoc.days

import com.capital7software.aoc.lib.AdventOfCodeTestBase
import org.junit.jupiter.api.Assertions.assertEquals
import org.junit.jupiter.api.BeforeEach
import org.junit.jupiter.api.Test
import org.slf4j.Logger
import org.slf4j.LoggerFactory

class Day23Test : AdventOfCodeTestBase() {
  companion object {
    val log: Logger = LoggerFactory.getLogger(Day23Test::class.java)
  }

  override fun getLogger(): Logger {
    return log
  }

  @BeforeEach
  fun setup() {
    val instance = Day23()
    setupFromFile(instance.defaultInputFilename)
  }

  @Test
  fun testGetValueInRegister() {
    val register = "a"
    val eggs = 7
    val instance = Day23()
    val expected = 3
    val actual = instance.getValueInRegister(lines, register, register, eggs)

    assertEquals(expected, actual, "The value in register a "
        + "is not what was expected: $expected")
  }

  @Test
  fun testGetValueInRegisterOptimized() {
    setupFromFile("inputs/input_day_23-02.txt")
    val register = "a"
    val eggs = 12
    val instance = Day23()
    val expected = 166
    val actual = instance.getValueInRegister(lines, register, register, eggs)

    assertEquals(expected, actual, "The value in register a "
        + "is not what was expected: $expected")
  }
}
